OptimizelySDK/Utils/CollectionExtensions.cs

```diff
@@ -0,0 +1,58 @@
+﻿/* 
+ * Copyright 2022, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OptimizelySDK.Utils
+{
+    public static class CollectionExtensions
+    {
+        /// <summary>
+        /// Merge a Dictionary into another without overwriting existing keys
+        /// </summary>
+        /// <param name="left">Dictionary that will be merged into</param>
+        /// <param name="right">Dictionary to merge into another</param>
+        /// <typeparam name="TKey">Key type</typeparam>
+        /// <typeparam name="TValue">Value type</typeparam>
+        /// <returns>Merged Dictionary</returns>
+        /// <exception cref="ArgumentNullException">Thrown if target Dictionary is null</exception>
+        public static Dictionary<TKey, TValue> MergeInPlace<TKey, TValue>(
```

Comment:
Can you please check if `Union` can be used.

Reply:
`.Union()` nor `.Concat()` with `.ToDictionary` doesn't seem to work.